apps/coscrad-frontend/src/store/slices/resources/shared/hooks/use-loadable-resources.ts

```diff
@@ -0,0 +1,59 @@
+import {
+    HttpStatusCode,
+    IBaseViewModel,
+    IHttpErrorInfo,
+    ResourceType,
+} from '@coscrad/api-interfaces';
+import { useSelector } from 'react-redux';
+import { ILoadable } from '../../../interfaces/loadable.interface';
+import { createResourceSelector } from '../selectors/create-resource-selector';
+
+type ResourceMap = Map<ResourceType, IBaseViewModel[]>;
+
+export const useLoadableResourcesMap = (): ILoadable<ResourceMap> => {
+    const resourceSelector = createResourceSelector();
+
+    const loadableResources = useSelector(resourceSelector);
+
+    const isLoading = Object.values(loadableResources).some(({ isLoading }) => isLoading);
+
+    const allErrorInfos = Object.values(loadableResources).reduce(
+        (accErrorInfos: IHttpErrorInfo[], { errorInfo }) =>
+            errorInfo ? accErrorInfos.concat(errorInfo) : accErrorInfos,
+        []
+    );
+
+    const compoundErrorInfo: IHttpErrorInfo =
+        allErrorInfos.length > 0
+            ? {
+                  code: HttpStatusCode.internalError,
+                  message: `Inner Errors: \n${allErrorInfos
+                      .map(({ message }) => message)
+                      .join('\n')}`,
+              }
+            : null;
+
+    if (isLoading || compoundErrorInfo)
+        return {
+            isLoading,
+            errorInfo: compoundErrorInfo,
+            // should we buble up what data we have so far?
```

Comment:
obviously a temp comment but (sp) buble

Reply:
Good catch. This file has since been removed.